### PR TITLE
[WIP] Added SympyProvider

### PR DESCRIPTION
**Pull request by yaelbh**
This pull request addresses Issue #9.

### CODE REVIEW

**Comment by jaygambetta** [comment]
this looks good. @nonhermitian and @diego-plan9 i think we need the register_external function now to go with this. 

`from qiskit_addon_sympy import SympyProvider`
`provider = SympyProvider()`
`register_external(provider)`

**Comment by yaelbh** [comment]
@jaygambetta I fixed the conflicts

File: examples/sympy_backends.py [hunk 1]
```diff
@@ -9,21 +9,29 @@
 amplitudes.
 """
 
-from qiskit_addon_sympy import StatevectorSimulatorSympy, UnitarySimulatorSympy
+import os
```
Comment (diego-plan9) [inline examples/sympy_backends.py]: Very minor - can you remove the `import osp` line, for keeping the example as tiny as possible, as it seems it is no longer used?

**Comment by yaelbh** [comment]
@diego-plan9  I see that you approved. However it is not clear to me if I should merge it now, or first wait for register_external and modify my code accordingly.

**Comment by diego-plan9** [comment]
Thanks @yaelbh ! Yes, some adjustments will be needed once we solve the registrations of addons in core - other than that, this PR is looking great, so merging!

**Pull request merged**